tests/unittests/core/test_primary_algo.py

```diff
@@ -161,8 +161,16 @@ def __init__(self, space: Space, fixed_suggestion: Trial):
         assert fixed_suggestion in space
 
     def suggest(self, num):
-        self.register(self.fixed_suggestion)
-        return [self.fixed_suggestion]
+        # NOTE: can't register the trial if it's already here. The fixed suggestion is always "new",
+        # but the algorithm actually observes it at some point. Therefore, we don't overwrite what's
+        # already in the registry.
+        if not self.has_suggested(self.fixed_suggestion):
+            self.register(self.fixed_suggestion)
+            return [self.fixed_suggestion]
+        # TODO: Does it make sense for the algorithm to just not suggest anything, if it has already
```

Comment:
We would expect the algorithm not to return duplicates, but the Producer is supposed to be able to handle duplicates if it ever happens. In this case since we are testing the algo only the correct behavior would be to return an empty list after first call to suggest. Does it mess up with some tests? It seems not right?

Reply:
Yeah it doesn't mess up the tests because the `fixed_suggestion` attribute is changed from the outside after each observed trial.

Reply:
Removed TODO in f492ffd5529907b1299db77c9ee403cccf39e612